extensions/shared/paid-blocks/utils.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { compact, get, startsWith } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { isSimpleSite } from '../site-type-utils';
+import getJetpackExtensionAvailability from '../get-jetpack-extension-availability';
+import getSiteFragment from "../get-site-fragment";
+
+/*
+ * Blocks list that require a paid plan.
+ */
+export const PAID_BLOCKS_LIST = [
+	'core/cover',
+	'core/video',
+];
+
+/**
+ * Return the checkout URL to upgrade the site plan,
+ * depending on the plan, postId, and postType site values.
+ *
+ * @param {object} siteParams -          Site params used to build the URL.
+ * @param {string} siteParams.planSlug - Plan slug.
+ * @param {string} siteParams.plan -     An object with details about the plan.
+ * @param {number} siteParams.postId -   Post id.
+ * @param {string} siteParams.postType - Post type.
+ * @returns {string}                     Upgrade URL.
+ */
+export function getUpgradeUrl( { planSlug, plan, postId, postType } ) {
```

Comment:
With `getUpgradeUrl` and `isUpgradable` being used in more than one component, should we move them into their own files into `extensions/shared` since they're being… shared?

Reply:
It's a good point and I've been thinking about this as well. Yeah..., agree. Let's move to a `share/plans-utils.js` file? File name is more than welcomed.